carto/maps_api.py

```diff
@@ -39,49 +40,46 @@ def send(self, url, http_method, http_header=None, file_body=None):
         :param file_body: The information in the json file needed to create the named map
         :return:
         """
-        if self.is_file is True:
-            data = self.client.send(url, http_method=http_method, http_headers=http_header, body=file_body)
-        else:
-            data = self.client.send(url, http_method=http_method, http_headers=http_header, json=file_body)
+        data = self.client.send(url, http_method=http_method, http_headers=http_header, body=file_body)
         data_json = self.client.get_response_data(data)
         self.update_from_dict(data_json)
 
-    def create(self):
+    def save(self):
         """
         Creates a new named map in the CARTO server
         :return: 
+
+        reconstruct json based on object parameters
         """
         header = {'content-type': 'application/json'}
-        if self.is_file is True:
-            with open(self.template_name) as payload:
-                self.send(self.api_url, "POST", header, payload)
+        attrs = [i for i in self.__dict__.keys() if i[:1] != '_']
```

Comment:
I was thinking, because we know the version of the template file schema (0.0.1), we could simply add something like (also note the changes for readability):

```python
if self.version == '0.0.1':
    attrs = ["version", "name", "auth", "placeholders", "layergroup", "view"]
else:
    attrs = [key for key in self.__dict__.keys() if not key.startswith('__')]